### PR TITLE
LPS-158706 Creation button for knowledge base root action

### PR DESCRIPTION
**Pull request by boton**
**Issue:** https://issues.liferay.com/browse/LPS-158706

**Done**
- [x] Add a custom empty state behind a feature flag
- [x] Add empty state actions
- [x] Add icons on the creation empty state button and in the Management toolbar
- [x] Avoid rendering a custom empty state if the view is a search empty result

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:relevant

**Review by adolfopa — APPROVED** [review]

**Comment by boton** [comment]
I updated the description after speaking with Maria. 
We are only left to hide the new empty state with action in the search view with 0 results.

File: modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/display/context/KBAdminManagementToolbarDisplayContext.java [hunk 1]
```diff
@@ -261,6 +261,83 @@ public CreationMenu getCreationMenu() throws PortalException {
 		return creationMenu;
 	}
 
+	public List<DropdownItem> getEmptyStateActionDropdownItems() {
+		long kbFolderClassNameId = PortalUtil.getClassNameId(
```
Comment (adolfopa) [inline modules/apps/knowledge-base/knowledge-base-web/src/main/java/com/liferay/knowledge/base/web/internal/display/context/KBAdminManagementToolbarDisplayContext.java:265]: @boton If you add here:
```
if (Validator.isNotNull(_getKeywords()) {
    return null;
}
```
it should do what you expect. I'm getting some weird errors when doing ant all, so I cannot test it myself at the moment.

**Review by adolfopa — CHANGES_REQUESTED** [review]

**Comment by boton** [comment]
Updated, rebased and ready!

@adolfopa I check if it is a search to avoid rendering a custom empty 2094bd2.

**Comment by boton** [comment]
ci:test:relevant

**Comment by boton** [comment]
ci:test:sf

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Pull request closed without merging**